### PR TITLE
Remove trailing whitespace in workflow generation

### PR DESCRIPTION
**Pull request by aqk**
This should be merged second

### CODE REVIEW

**Review by hoffmang9 — DISMISSED** [review]
Nice.

**Comment by xdustinface** [comment]
Hmm #7053 already takes care of this, no? 😄

**Comment by xdustinface** [comment]
Or whats is the "Regenrate workflow" and adding this extra space there in between about?

**Comment by hoffmang9** [comment]
> Or whats is the "Regenrate workflow" and adding this extra space there in between about?

The majority of the CI test workflows are actually generated by a script @aqk made. That script was outputting the trailing spaces in the CI .ymls that the hooks were mostly finding.

**Comment by xdustinface** [comment]
> > Or whats is the "Regenrate workflow" and adding this extra space there in between about?
> 
> The majority of the CI test workflows are actually generated by a script @aqk made. That script was outputting the trailing spaces in the CI .ymls that the hooks were mostly finding.

Ahh, i see. But now this PR removes the trailing but adds an extra whitespace between those `-v   --durations` which is probably not by intention @aqk?

**Comment by aqk** [comment]
@xdustinface It is, in fact, intentional.

**Comment by xdustinface** [comment]
Okay got it, can you elaborate the intention? I mean, at the end its really not an issue at all since its only an extra whitespace :smile:  but im still curious about the idea behind it.

**Comment by aqk** [comment]
Sure! The way this works is, it is a simple templating engine that replaces strings in the input document with generated strings from `build-workflows.py`, to work around limitations in pytest and the github CI.

The line in question that produces the extra whitespace is:

`./venv/bin/py.test TEST_DIR -s -v --durations 0 PYTEST_PARALLEL_ARGS`

In most cases, `PYTEST_PARALLEL_ARGS` is empty, so we see one extra whitespace character at the end of the line.

Now, the right way to handle this is for the template document and the template engine to be aware of whether the variable should be surrounded by whitespace or not.

The reason it is "intentional" above is not because I want an extra whitespace, it is because I was making a decision between the value of making the simple change to the data, or dealing with the extra complexity of generalizing a script that generates these workflow files. I decided it was not worth it.

By the way, if you think a change is important, just say "hey, I think we should not have these whitespaces". Then I can say "Ah, that seems right", or "no, the reason is xyz". It is okay.

File: .github/workflows/build-test-ubuntu-core-full_node.yml [hunk 1]
```diff
@@ -97,14 +97,9 @@ jobs:
     - name: Install developer requirements
       run: |
         . ./activate
-        venv/bin/python -m pip install pytest pytest-asyncio pytest-xdist pytest-monitor
```
Comment (arvidn) [inline .github/workflows/build-test-ubuntu-core-full_node.yml:100]: it looks like you dropped this dependency, along with the code using it below. Are these files generated and I added code to it manually?

File: .github/workflows/build-test-ubuntu-core-full_node.yml [hunk 2]
```diff
@@ -97,14 +97,9 @@ jobs:
...
 
     - name: Test core-full_node code with pytest
       run: |
         . ./activate
-        ./venv/bin/py.test tests/core/full_node/test_address_manager.py tests/core/full_node/test_block_store.py tests/core/full_node/test_coin_store.py tests/core/full_node/test_conditions.py tests/core/full_node/test_full_node.py tests/core/full_node/test_full_node_store.py tests/core/full_node/test_initial_freeze.py tests/core/full_node/test_mempool.py tests/core/full_node/test_mempool_performance.py tests/core/full_node/test_node_load.py tests/core/full_node/test_performance.py tests/core/full_node/test_sync_store.py tests/core/full_node/test_transactions.py -s -v --durations 0 
-    - name: Check resource usage
-      run: |
-        sqlite3 -readonly -separator " " .pymon "select item,cpu_usage,total_time,mem_usage from TEST_METRICS order by mem_usage desc;" >metrics.out
-        cat metrics.out
-        ./tests/check_pytest_monitor_output.py <metrics.out
```
Comment (arvidn) [inline .github/workflows/build-test-ubuntu-core-full_node.yml:110]: I take it removing this was unintentional

**Review by arvidn — CHANGES_REQUESTED** [review]

**Comment by arvidn** [comment]
the parts of the workflow that's being dropped by this PR is also being changed here: https://github.com/Chia-Network/chia-blockchain/pull/6972

File: .github/workflows/build-test-ubuntu-core-full_node.yml (hunk 1 above)
Comment (aqk) [reply .github/workflows/build-test-ubuntu-core-full_node.yml:100]: Yes, these are generated - I will add a warning to the generated files

**Comment by aqk** [comment]
@arvidn https://github.com/Chia-Network/chia-blockchain/pull/7181

**Comment by aqk** [comment]
Please merge https://github.com/Chia-Network/chia-blockchain/pull/7181 first.

File: .github/workflows/build-test-ubuntu-core-full_node.yml (hunk 2 above)
Comment (aqk) [reply .github/workflows/build-test-ubuntu-core-full_node.yml:109]: Thanks for pointing this out

**Pull request closed without merging**